src/screens/VeggieScreen/NewVeggieLogModal.tsx

```diff
@@ -20,25 +21,26 @@ export const NewVeggieLogModal = ({
   const [calendarVisible, setCalendarVisible] = useState(false);
   const [date, setDate] = useState(Date.now());
   const [notes, setNotes] = useState("");
-  const { veggieId } = route.params;
+  const { veggieId, locationTitles } = route.params;
   const [payloadTags, setPayloadTags] = useState<Tag[]>([]);
 
   const appContext = useContext(pressedTagsContext);
   const pressedTags = appContext?.pressedTags;
   const setPressedTags = appContext?.setPressedTags;
+  const location: LocationObj = locationTitles;
 
   const dispatch = useAppDispatch();
   const cachedPhotos = useAppSelector(photoSelectors.selectAllCachedPhotos);
 
   useEffect(() => {
     setPayloadTags([...pressedTags!]);
   }, [pressedTags]);
-
   const handleSubmit = () => {
     const { payload } = dispatch(
       logActions.add({
         veggie: veggieId,
         date,
+        location,
```

Comment:
the above location variable is really just passing on the variable and could be removed
```suggestion
        location: locationTitles,
```